c/driver/postgresql/copy/writer.h

```diff
@@ -92,13 +92,20 @@ class PostgresCopyFieldWriter {
  public:
   virtual ~PostgresCopyFieldWriter() {}
 
-  void Init(struct ArrowArrayView* array_view) { array_view_ = array_view; };
+  template <class T, typename... Params>
+  static std::unique_ptr<T> Create(struct ArrowArrayView* array_view, Params&&... args) {
```

Comment:
I don't think this even needs to be a static method? It could just be a free function and you can then just invoke

```cpp
*out = MakeWriter<T>(array_view);
```

Reply:
I think would have to make the Init method public to do that - do you think that is better?

```cpp
../driver/postgresql/copy/writer.h:564:15: error: ‘virtual void adbcpq::PostgresCopyFieldWriter::Init(ArrowArrayView*)’ is protected within this context
  564 |   writer->Init(array_view);
      |   ~~~~~~~~~~~~^~~~~~~~~~~~
../driver/postgresql/copy/writer.h:100:16: note: declared protected here
  100 |   virtual void Init(struct ArrowArrayView* array_view) { array_view_ = array_view; };
```

Reply:
Ah...

This is fine then. The `T::Create<T>` bugged me a little bit but it's not actually a problem.